p2p/server/msgserver.go

```diff
@@ -49,10 +50,11 @@ func NewMsgServer(network Service, name string, requestLifetime time.Duration, l
 		resHandlers:        make(map[uint64]func(msg []byte)),
 		pendingQueue:       list.New(),
 		network:            network,
-		ingressChannel:     network.RegisterProtocol(name),
+		ingressChannel:     network.RegisterProtocolWithChannel(name, make(chan service.Message, 100)),
```

Comment:
change to NewMsgServer(network Service, name string, requestLifetime time.Duration,c chan service.Message ,logger log.Log)
and pass c to RegisterProtocolWithChannel

Reply:
And what about workerLimiter? Do I pass an int or just leave it hardcoded?

Reply:
And what about workerLimiter? Do I pass an int or just leave it hardcoded?

Reply:
you can leave the worker limit as it is

Reply:
actually would be better to use runtime.numcpu() for the worker limit